engine/repositories/processor_push.go

```diff
@@ -0,0 +1,99 @@
+package repositories
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+func (s *Service) processPush(op *sdk.Operation) error {
+	gitRepo, path, currentBranch, err := s.processGitClone(op)
+	if err != nil {
+		return sdk.WrapError(err, "unable to process gitclone")
+	}
+	//Check is repo has diverged
+	hasDiverged, err := gitRepo.HasDiverged()
+	if err != nil {
+		log.Error("Repositories> processPush> HasDiverged> [%s] Error: %v", op.UUID, err)
+		return err
```

Comment:
In this func many errors needs to be wrapped sdk.WithStack(err)

Reply:
don't need i think